api/src/main/java/org/pmiops/workbench/db/dao/UserRecentResourceServiceImpl.java

```diff
@@ -50,11 +56,7 @@ public UserRecentResource updateNotebookEntry(long workspaceId, long userId, Str
     UserRecentResource recentResource = getDao().findByUserIdAndWorkspaceIdAndNotebookName(userId, workspaceId, notebookNameWithPath);
     if (recentResource == null) {
       handleUserLimit(userId);
-      recentResource = new UserRecentResource();
-      recentResource.setUserId(userId);
-      recentResource.setWorkspaceId(workspaceId);
-      recentResource.setCohort(null);
-      recentResource.setNotebookName(notebookNameWithPath);
+      recentResource = new UserRecentResource(workspaceId, userId, notebookNameWithPath, lastAccessDateTime);
```

Comment:
I think you could also use the builder pattern here and below (Greg just taught me this one!)
```
recentResource = new UserRecentResource()
    .workspaceId(workspaceId)
    .userId(userId)
    .notebookName(notebookNameWithPath)
    .lastAccessDate(lastAccessDateTime)
```
Or something? And that would avoid the multiple defined constructors.

Reply:
Sorry @blrubenstein - our entities are not built that way :-(  Model classes derived from swagger get that pattern for free, so wherever you use one of those, you can use that pattern.

Reply:
Oh no... okay... :(

Reply:
FYI it also makes me sad that they aren't symmetric as I find this pattern much more readable. I filed https://precisionmedicineinitiative.atlassian.net/browse/RW-1384 a few weeks ago to apply this pattern throughout our models (Cohort Builder models currently do have chaining, actually).